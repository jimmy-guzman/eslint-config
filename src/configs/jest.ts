import type { TypedConfigItem } from "../types";

import { GLOB_E2E, GLOB_TESTS } from "../constants";
import { jestRules } from "../rules/jest";
import { interopDefault } from "../utils/interop-default";

export default async function jestConfig() {
  const jestPlugin = await interopDefault(import("eslint-plugin-jest"));

  return [
    {
      files: GLOB_TESTS,
      ignores: GLOB_E2E,
      ...jestPlugin.configs["flat/recommended"],
      name: "jimmy.codes/jest",
      rules: await jestRules(),
    },
  ] satisfies TypedConfigItem[];
}
