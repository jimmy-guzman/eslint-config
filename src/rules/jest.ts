import { type Linter } from "eslint";
import jest from "eslint-plugin-jest";

export const jestRules = {
  ...jest.configs["flat/recommended"].rules,
  ...jest.configs["flat/style"].rules,
  "jest/consistent-test-it": [
    "error",
    {
      fn: "test",
      withinDescribe: "it",
    },
  ],
} satisfies Linter.RulesRecord;
