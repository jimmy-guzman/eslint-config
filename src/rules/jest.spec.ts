import { jestRules } from "./jest";

test("should create jest rules", () => {
  expect(jestRules).toMatchSnapshot();
});
