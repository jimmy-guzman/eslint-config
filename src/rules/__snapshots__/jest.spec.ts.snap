// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should create jest rules 1`] = `
{
  "jest/consistent-test-it": [
    "error",
    {
      "fn": "test",
      "withinDescribe": "it",
    },
  ],
  "jest/expect-expect": "warn",
  "jest/no-alias-methods": "warn",
  "jest/no-commented-out-tests": "warn",
  "jest/no-conditional-expect": "error",
  "jest/no-deprecated-functions": "error",
  "jest/no-disabled-tests": "warn",
  "jest/no-done-callback": "error",
  "jest/no-export": "error",
  "jest/no-focused-tests": "error",
  "jest/no-identical-title": "error",
  "jest/no-interpolation-in-snapshots": "error",
  "jest/no-jasmine-globals": "error",
  "jest/no-mocks-import": "error",
  "jest/no-standalone-expect": "error",
  "jest/no-test-prefixes": "error",
  "jest/prefer-to-be": "error",
  "jest/prefer-to-contain": "error",
  "jest/prefer-to-have-length": "error",
  "jest/valid-describe-callback": "error",
  "jest/valid-expect": "error",
  "jest/valid-expect-in-promise": "error",
  "jest/valid-title": "error",
}
`;
